util/dockerutil/client.go

```diff
@@ -63,6 +63,28 @@ func (c *Client) LoadImage(ctx context.Context, name string, status progress.Wri
 	}, nil
 }
 
+func (c *Client) Features(ctx context.Context, name string) (map[Feature]bool, error) {
```

Comment:
In discussion with @crazy-max about https://github.com/docker/buildx/pull/1846, we noticed that this probably needs a similar cache to ensure that we don't create a new docker client in each loop of the `toSolveOpt` call.

Reply:
Can try and do this as a follow-up though, would be good to not block another RC on this.